lib/instrumentation/index.js

```diff
@@ -59,6 +59,50 @@ function Instrumentation (agent) {
       return this.bindingSpan || this.activeSpan
     }
   })
+
+  // NOTE: we need to track module names for patches
+  // in a separate array rather than using Object.keys()
+  // because the array is given to the hook(...) call.
+  this._modules = []
```

Comment:
Because that will only be the _original_ set of module names. This PR works by _kind of_ a hack that pushing new items into the array already given to `hook(...)` will allow new patches to be added after the hook has already been attached.

Reply:
But we don't allow the user to call `registerPatch` after `hook()` have been called anyway right?